src/transformState.js

```diff
@@ -5,7 +5,19 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  actions.forEach((action) => {
+    if (action.type === 'addProperties') {
```

Comment:
Generally avoid else if, else if, else if conditional chaining, [switch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) would be better here. 

For example you want to execute the last condition for 'clear', javascript will go and check conditions one by one before finally executing the last, correct one. If there were more this could take a while.

In the switch statement on the other hand, compiler creates a jump table and immediately "jumps" into the right clause rather than checking conditions / clauses one by one.

Reply:
Yea that's true I will use switch next time :D